### ISSUE TITLE
Permission bug: Bluetooth scanning without location permission on Android 12+

### ISSUE THREAD

**Issue by janusw** [dotnet/maui#20871]
### Description

Since Android 12, it is possible to do Bluetooth scanning without location permissions. To achieve this, one has to declare the BT scanning permission in the Android manifest like this:

```
<uses-permission android:name="android.permission.BLUETOOTH_SCAN" android:usesPermissionFlags="neverForLocation" />
```

Requesting the permissions at runtime works well for this scenario, as long as `ACCESS_FINE_LOCATION` is not declared in the manifest:

```
            PermissionStatus status = await Permissions.CheckStatusAsync<Permissions.Bluetooth>();

            if (status != PermissionStatus.Granted)
                status = await Permissions.RequestAsync<Permissions.Bluetooth>();
```

However, if `ACCESS_FINE_LOCATION` is declared in the manifest, but is denied at runtime (while `BLUETOOTH_SCAN` is granted), then the snippet above reports the status as `PermissionStatus.Denied`.

This wrongly signals to the application that scanning is not possible, while in fact it is.

### Steps to Reproduce

1. Check out the `dotnet-bluetooth-le` repository (link below).
2. Build the sample `BLE.Client.MAUI` and run it on a Android 12+ device.
3. Reject the location permission, but accept the Bluetooth/Nearby devices permission at runtime.
4. As a result, the app will not start scanning (although it could): "Permissions denied. Not scanning."

### Link to public reproduction project repository

https://github.com/dotnet-bluetooth-le/dotnet-bluetooth-le

### Version with bug

8.0.7 SR2

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 12 and up

### Did you find any workaround?

Ignore the wrong permission status.

**Comment by janusw** [comment]
Btw, the exact same problem also affects `Permissions.NearbyWifiDevices`, which can also be used with `android:usesPermissionFlags="neverForLocation"`.

The permissions `Bluetooth` and `NearbyWifiDevices` were introduced with PR #12413 and #14718.

Android documentation references:
* https://developer.android.com/develop/connectivity/bluetooth/bt-permissions#assert-never-for-location
* https://developer.android.com/develop/connectivity/wifi/wifi-permissions#assert-never-for-location

**Comment by bernieEV** [comment]
YES, I want measurement equipment using bluetooth and smartphone as control device/display never ask for location permission!

**Comment by janusw** [comment]
> In the same app on 3 separate pages
> 
>     * I need the bluetooth to connect to low energy device,
> 
>     * need location permission to display on map for a feature
> 
> when I added bluetooth permission in manifest, I started to get the error about needing to declare fine location permission, even though it already declared.

@walnut-co Yes, this is exactly the MAUI permission bug that is described by this issue and fixed by PR #20874 (which unfortunately has still not been approved yet):
* Up to Android API level 30, `ACCESS_FINE_LOCATION` was required for Bluetooth functionality.
* Starting from API level 31 it is optional, if `neverForLocation` is specified (as you did).

Your manifest is fine. MAUI should allow you to use Bluetooth without (fine-grained) location permissions on recent Android versions (12 and newer). Once #20874 is approved, this should work well.

**Comment by janusw** [comment]
> Once [#20874](https://github.com/dotnet/maui/pull/20874) is approved, this should work well.

Please give a heart or thumbs-up reaction there to signal support for this PR.

**Comment by janusw** [comment]
Wow, what a ride for an innocent patch. Thanks for finally getting this in, @PureWeen ! 🥳 

Since PR #20874 has the milestone ".NET 9 SR10", I assume the fix will be included in the next MAUI 9 release (probably 9.0.100) ...
